README.md

```diff
@@ -449,6 +449,68 @@ Note that because this method is not available in the generic `IDataReader` inte
 TimeSpan timeSpanTime = ((SnowflakeDbDataReader)reader).GetTimeSpan(13);
 ```
 
+Executing a Batch of SQL Statements (Multi-Statement Support)
+--------------------------------------------------------------
+
+With the .NET 2.0.18 and later connector, you can send
+a [batch of SQL statements](https://docs.microsoft.com/en-us/sql/odbc/reference/develop-app/batches-of-sql-statements?view=sql-server-ver15)  (separated by semicolons)
+to be executed in a single request. For example:
+
+```cs
+IDbCommand cmd = conn.CreateCommand();
+                var param = cmd.CreateParameter();
+                param.ParameterName = "MULTI_STATEMENT_COUNT";
+                param.DbType = DbType.Int16;
+                param.Value = 3;
+                cmd.Parameters.Add(param);
+                cmd.CommandText = "select 1; select 2, 3; select 4, 5, 6";
+                IDataReader reader = cmd.ExecuteReader();
+```
+
+To send a batch of statements with the Snowflake .NET connector, you must specify the number of statements in the batch.
+The Snowflake database requires the exact number of statements in order to guard against
+[SQL injection](https://en.wikipedia.org/wiki/SQL_injection>) attacks. 
+
+The next section explains how to specify the number of statements in a batch. 
+
+### Specifying the Number of Statements in a Batch
+
+By default, the Snowflake database expects a .NET connector application to prepare and send a single statement for execution.
+
+You can override this by specifying the number of statements in a batch for a given request or by enabling multiple statements for
+the current session.
+
+To enable multiple statements for the current session or account, alter the session or account, and set the Snowflake [MULTI_STATEMENT_COUNT](https://docs.snowflake.com/en/sql-reference/parameters.html#label-parameters-multi-statement-count) parameter to ``0``, as follows.
+
+```
+alter session set MULTI_STATEMENT_COUNT = 0;
```

Comment:
This actually conflicts with the previous statement ("you must specify the number of statements in the batch. The Snowflake database requires the exact number of statements in order to guard against SQL injection attacks").

I think this is supposed to say that you should set MULTI_STATEMENT_COUNT to the number of statements in the batch.

Reply:
Not quite. I also removed the mention of account (as .NET does not support it). The idea is that you must enable the feature by setting the param to 0. Not a good design choice IMHO, but the values are 0 (multiple) or 1 (single). It has nothing to do with the specific number of statements.

Reply:
So this statement in line 471 is not correct:

```
The Snowflake database requires the exact number of statements
```

0 is not the exact number of statements.

Reply:
It requires the exact number in the call. This parameter simply enables or disables the ability to send more than one statement. Unfortunately, it is also backwards as 0 means yes and 1 means no.

Reply:
So the session is created and started by the .NET connector, right?
I write code that uses the .NET connector.
How in my code do I set the parameter for the session?

As an example, [in this test](https://github.com/snowflakedb/snowflake-connector-net/blob/e9673a50a3c69bb6a2c7807cf426699b253df9df/Snowflake.Data.Tests/SFMultiStatementsIT.cs#L34), I'm opening a connection, which I think creates the session.

Where in this code would I do the equivalent of `ALTER SESSION SET MULTI_STATEMENT_COUNT = 0`?

Reply:
Re-reading this again, I think the problem is that there's not enough of an explanation of how to do this per request. When I skimmed this section, the most prominent thing was that I should execute ALTER SESSION somewhere.

I think we need to make it easier to see that there are two options here. I think adding an example and more of an explanation of first option would help. Otherwise, to someone skimming this section, the takeaway is that you just use ALTER SESSION to set MULTI_STATEMENT_COUNT to 0.

Reply:
Here's a suggested way to do this.

```
To send a batch of statements with the Snowflake .NET connector, you can either:

- Specify the number of statements in the batch, or
- Override the `MULTI_STATEMENT_COUNT` parameter for the session to indicate that each batch contains more than one statement.

### Specifying the Number of Statements in a Batch

By default, the Snowflake database expects a .NET connector application to prepare and send a single statement for execution.

You can override this by specifying the number of statements in a batch for a given request.
The Snowflake database requires the exact number of statements in order to guard against
[SQL injection](https://en.wikipedia.org/wiki/SQL_injection>) attacks. 

To specify the number of statements in the batch, set the `MULTI_STATEMENT_COUNT` parameter. For example, if there are two statements in the batch:

    var param = cmd.CreateParameter();
    param.ParameterName = "MULTI_STATEMENT_COUNT";
    param.DbType = DbType.Int16;
    param.Value = 2;
    cmd.Parameters.Add(param);

### Enabling Multiple Statements in a Session

If you don't want to specify the number of statements in each request, you can enable multiple statements for the current session by setting the Snowflake [MULTI_STATEMENT_COUNT](https://docs.snowflake.com/en/sql-reference/parameters.html#label-parameters-multi-statement-count) parameter to ``0``, as follows.

   <example of .NET connector code that sets this>
```

Reply:
Otherwise, the code block is the most prominent element in the section, and [it's easy to miss the fact that the code block just covers the second of two options](https://drive.google.com/a/snowflake.com/uc?export=view&id=1ysK5FoX8_DbeAkJPupTF4Pbkzdwl-fqo).